airflow/api_connexion/endpoints/task_endpoint.py

```diff
@@ -22,21 +22,16 @@
 from airflow.api_connexion import security
 from airflow.api_connexion.exceptions import BadRequest, NotFound
 from airflow.api_connexion.schemas.task_schema import TaskCollection, task_collection_schema, task_schema
+from airflow.auth.managers.models.resource_details import DagAccessEntity
 from airflow.exceptions import TaskNotFound
-from airflow.security import permissions
 from airflow.utils.airflow_flask_app import get_airflow_app
 
 if TYPE_CHECKING:
     from airflow import DAG
     from airflow.api_connexion.types import APIResponse
 
 
-@security.requires_access(
-    [
-        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
-        (permissions.ACTION_CAN_READ, permissions.RESOURCE_TASK_INSTANCE),
-    ],
-)
+@security.requires_access_dag("GET", DagAccessEntity.TASK_INSTANCE)
```

Comment:
Is this right? Task is slightly differently from task instance since the latter also requires permission on DAG run.

Reply:
Can you elaborate? This is the exact same permissions we were checking before. Before we were checking permissions on task instance and dag level. `@security.requires_access_dag("GET", DagAccessEntity.TASK_INSTANCE)` does exactly the same

Reply:
The task endpoints checked for DAG and task instance perms, while the task instance ones checked for them plus DAG run perms. Now they all check for the same perms.

Reply:
Done :)